book/Zapp-Pipes/protocols/login_and_payment_protocol.md

```diff
@@ -0,0 +1,46 @@
+# Login and Payment Protocol
+
+This document outlines the requirements for mapping the data to determine whether login and/or payment are required to gain access to a video item.
+
+## The "Free" flag
+Previously, the flag that was used to achieve this was the "free" flag in the extensions section of the item. However, there are a few drawbacks for using the "free" flag:
+1. The flag doesn't distinguish between whether an item requires login or payment to get access to, and might create some confusion. 
+2. If the item requires purchase, the "free" flag doesn't specify what entitlements are needed in order to unlock the item.
+
+The suggested protocol is meant to be used by DSPs, who want to provide this information to the app, and can be used by Login / Payment plugins to interpret the action that is needed for the item to be unlocked. 
+
+**Important Note**: if you are using Applicaster2 as your datasource provider, there is no way to add the new extensions to an item, and therefore the "free" flag is the only way to configure an item as "locked".
+
+
+## "requires_authentication" and "ds_product_ids"
+
+The protocol is based on two fields that are added to the "extensions" section of the item:
+
+* **requires_authentication**: a boolean flag to determine whether login is required to unlock the item. 
+This field should be of type "boolean" (true/false) and *not* a "string".
+If the flag is not specified, the default value is `false`, which means the item does not require authentication.
+
+* **ds_product_ids**: a list of entitlements needed to unlock the item. 
+It is up to the Login / Payment provider to decide if *all* entitlements are required, or a single entitlement is sufficient. 
```

Comment:
Oh really? I didn't know that. I thought one of the entitlements would suffice. How would it work if two entitlements are required?

Reply:
Maybe it's not a valid case, and the only valid case is a "OR" between the entitlements. I didn't want to limit that, because technically it's not the DSP's concern how the entitlements data is used. I would change the text to say something "typically a single entitlement fulfilled should give access to the item"

Reply:
xI am not referring to this anymore in the document